.github/workflows/test.yml

```diff
@@ -2,28 +2,50 @@ name: test
 on:
 - pull_request
 jobs:
-  console-kit_xenial:
-    container: 
-      image: vapor/swift:5.2-xenial-ci
+  console-kit_linux:
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        image: ['swift:5.2-xenial', 'swift:5.2-bionic',
+                'swiftlang/swift:nightly-5.2-xenial', 'swiftlang/swift:nightly-5.2-bionic',
+                'swiftlang/swift:nightly-5.3-xenial', 'swiftlang/swift:nightly-5.3-bionic',
+                'swiftlang/swift:nightly-master-xenial', 'swiftlang/swift:nightly-master-bionic',
+                'swiftlang/swift:nightly-master-focal', 'swiftlang/swift:nightly-master-centos8',
+                'swiftlang/swift:nightly-master-amazonlinux2']
+    container: ${{ matrix.image }}
     steps:
-    - uses: actions/checkout@v1
-    - run: swift test --enable-test-discovery --sanitize=thread
-  console-kit_bionic:
-    container: 
-      image: vapor/swift:5.2-bionic-ci
-    runs-on: ubuntu-latest
-    steps:
-    - uses: actions/checkout@v1
-    - run: swift test --enable-test-discovery --sanitize=thread
-  vapor:
-    container: 
-      image: vapor/swift:5.2-ci
-    runs-on: ubuntu-latest
+      - run: dnf install -y zlib-devel
+        if: ${{ matrix.image == 'swiftlang/swift:nightly-master-centos8' }}
+      - run: yum install -y zlib-devel
+        if: ${{ matrix.image == 'swiftlang/swift:nightly-master-amazonlinux2' }}
+      - uses: actions/checkout@v2
+        with:
+          path: console-kit
+      - uses: actions/checkout@v2
+        with:
+          repository: vapor/vapor
```

Comment:
I think it could make sense to have this as a separate test still. We don't really need to compile and test Vapor on every platform. Just one is good enough to make sure we didn't break API or introduce errors. Anything potentially different between operating systems should already be tested within ConsoleKit's unit tests.

Reply:
Yeah, I was thinking that myself - but at the same time it's running within the same job so it's not using up extra slots (whereas a separate test would, ironically), and it provides a not-unwelcome backup check of Vapor itself in case the CI workflows don't all match up (and thus far I don't think they ever have yet 😅).

Reply:
I think having just one job compiling Vapor and running its unit tests instead of all 12+ jobs doing it would mean CI would run faster. Waiting on CI is the bane of my existence and kills my productivity so anything to help mitigate that is greatly appreciated. If I'm wrong and it wouldn't actually be faster then I'm fine with how it is.

Reply:
I mean, is it faster to build just `console-kit` than it is to build `vapor`? Sure. Is the difference particularly significant when using this "all at once" setup instead of the previous "do console-kit and then separately do vapor+console-kit" way? From what I've seen, the GitHub runners are powerful enough that the extra runtime measures around 10 seconds per job. CI via GH Actions takes most of its time in setup/teardown overhead, so the Vapor build and tests would have take about 3x as long as they do now for the difference to matter much.

Reply:
![Screen Shot 2020-05-20 at 11 55 00 AM](https://user-images.githubusercontent.com/1342803/82468433-c936c780-9a90-11ea-9d79-822187dd510e.png)

From that it seems like testing Vapor is adding a whopping 3 minutes to each test case. The total runtime is 5 minutes so it's more than doubling it. 

Do you mean _total_ runtime is only increased by 10 seconds?

Reply:
TBH, off the top of my head I don't actually recall which number I was looking at. Either way, I'm fine with running just one set of Vapor tests, definitely beats going back and forth over it like this 🙂 Will update the PR shortly.